packages/labs/test-projects/test-element-a/src/element-events.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {LitElement, html} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
+import {SpecialEvent} from './special-event.js';
+import {MyDetail} from './detail-type.js';
+
+export {SpecialEvent} from './special-event.js';
+export {MyDetail} from './detail-type.js';
+export class EventSubclass extends Event {
+  aStr: string;
+  aNumber: number;
+
+  constructor(
+    aStr = 'aStr',
+    aNumber = 5,
+    type = 'event-subclass',
+    options = {composed: true, bubbles: true, cancelable: true}
+  ) {
+    super(type, options);
+    this.aStr = aStr;
+    this.aNumber = aNumber;
+  }
+}
+
+declare global {
+  interface HTMLElementEventMap {
+    'event-subclass': EventSubclass;
+    'special-event': SpecialEvent;
+    'string-custom-event': CustomEvent<string>;
+    'number-custom-event': CustomEvent<number>;
+    'my-detail-custom-event': CustomEvent<MyDetail>;
+  }
+}
+
+/**
+ * My awesome element
+ * @fires string-custom-event {CustomEvent<string>} A custom event with a string payload
+ * @fires number-custom-event {CustomEvent<number>} A custom event with a number payload
+ * @fires my-detail-custom-event {CustomEvent<MyDetail>} A custom event with a MyDetail payload.
```

Comment:
Can we add a 
```ts
* @fires template-result-custom-event {CustomEvent<TemplateResult>}
```
and dispatch an event like 
```ts
    fromNode.dispatchEvent(new CustomEvent('template-result-custom-event', {detail: html``}))
```
so that we can exercise type references from an external package that's not otherwise imported into the generated source?

Note that this will point out an obvious catch-22: I expect the `TemplateResult` reference in the `@fires` will cause the analyzer to error because `TemplateResult` symbol isn't imported into the current file, which I believe is the right thing to do, since we can't otherwise know what that identifier was actually referring to. But adding an `import type {TemplateResult} from 'lit';` will fail compilation because it's not actually used in the TS source, only the comment.

We can either silence the unused symbol warning, or (better imo) just require the user to actually use the imported symbol, ala `new CustomEvent('template-result-custom-event', {detail: html`` as TemplateResult})`.

Either way, this will allow us to test that we get the proper `import {TemplateResult} from 'lit'` in the generated source.

Reply:
Done.